### DISCUSSION TITLE
Gatsby Head API and client-only routes

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [gatsbyjs/gatsby#36551] [Help]
I have a client-only route with `param.id`. This ID is used to fetch client-only data using Apollo Client.

Now I want to implement the Gatsby Head API; but the `head` function executes outside the root element, so my graph provider (and hence my Apollo Client) are unavailable.

How might I safely work around this gap? Initializing Apollo Client all over again is not a good solution, but I'm hoping there might be some other way.

**Comment by marvinjude** [comment]
Hi, @aaronadamsCA,

In the future [`WrapRootElement`](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#wrapRootElement) will make this possible out of the box so you can consume context from `Head`. Asides from reinitializing the Apollo Client, I can't really think of a workaround for this.

### Related comment

https://github.com/gatsbyjs/gatsby/discussions/35841#discussioncomment-3256204

**Reply by aaronadamsCA** [reply]
Thanks @marvinjude. I thought about grabbing an additional `ApolloClient`, but as soon as I realized this would also need to consume site config, I realized the size of the change would get rough pretty quickly.

Is this improvement imminent? I am not looking for a guarantee, but I need to weigh the impact of waiting for this vs. switching to `react-helmet-async`.

**Reply by marvinjude** [reply]
> Is this improvement imminent?

We have this planned for v5. So it's _soonish_.

**Reply by JustFly1984** [reply]
@marvinjude I do not see any @5-alpha versions in npm, nor any v5 branches in gatsby repository. Can we expect to get build fixed by this month?
Can you provide any workaround? 

I need to deploy to QA asap.

**Reply by LekoArts** [reply]
We can't give any timelines on releases for v5 yet, you'll see the announcements on all relevant channels when v5 is ready to try and it includes the feature. Please use `react-helmet-async` so long then if you need this functionality.
